ccswbs/gus
# Switch from BASIC_AUTH to API_KEY

# Summary of changes
In preparation for the backend upgrade to Drupal 10, switch from using BASIC_AUTH environment variables to API_KEY for better security and stability and to ensure menus are pulled in correctly. See https://www.drupal.org/project/key_auth and https://www.drupal.org/docs/contributed-modules/gatsby-integration/2x-module-setup-guide#s-gatsby-application for more info.

**Note:** The API_KEY environment variable is already present on both the Netlify and Azure build environments. Once this PR is approved and merged, the BASIC_AUTH environment variables can be removed.

## Frontend
- Delete BASIC_AUTH environment variables from `gatsby-config.js`
- Add API_KEY to `gatsby-config.js`

### Incidental changes
- Upgrade packages to latest versions to ensure maximum compatibility with Drupal 10
- Add graphiql explorer to package.json
- Add `drupal_internal__tid` to `taxonomy_term__event_categories` schema to resolve local build errors

## Backend
- The key_auth module was already added to the Drupal backend as per [an earlier multidev merge](https://teams.microsoft.com/l/message/19:f2ae5b00dc95407d8902b0e04ff98ec1@thread.tacv2/1698260331115?tenantId=be62a12b-2cad-49a1-a5fa-85f4f3156a7d&groupId=cde949b1-e9d8-483b-8d20-e9b8fd9e493d&parentMessageId=1698260331115&teamName=Content%20Hub%20Team&channelName=Multidev%20Merge%20Requests&createdTime=1698260331115&allowXTenantAccess=false)
- If you want to test locally with the Drupal 10 multidev, use https://conversion-chug.pantheonsite.io/ as base_url

# Test Plan
- Use `npm ci` and then run a local build
    - Ensure `gatsby develop` completes successfully
    - Ensure [graphiql](http://localhost:8000/___graphql) loads and functions as expected
- Review Netlify build at https://keyauth--drupal10test.netlify.app/ (uses D10 conversion site as backend)
- Review Netlify PR build at https://deploy-preview-178--ugconthub.netlify.app/ (this one uses api.live as backend)
